CONTRIBUTING.md

```diff
@@ -0,0 +1,69 @@
+# Contributing
+
+## Getting Started
+
+General information about setting up your Python environment, testing modules,
+Ansible coding styles, and more can be found in the [Ansible Community Guide](
+https://docs.ansible.com/ansible/latest/community/index.html).
+
+Information about boto library usage, module utils, testing, and more can be
+found in the [AWS Guidelines](https://docs.ansible.com/ansible/devel/dev_guide/platforms/aws_guidelines.html)
+documentation.
+
+## AWS Collections
+
+There are two related collections containing AWS content (modules and plugins).
+
+### amazon.aws
+This collection contains the `module_utils` (shared libraries) used by both collections.
+Content in this collection is included downstream in Red Hat Ansible Automation Platform.
+
+Code standards, test coverage, and other supportability criteria may be higher in this collection. 
+
+### community.aws
+This collection contains modules and plugins contributed and maintained by the Ansible AWS
+community.  The `community.aws` collection is tested and generally assured to work in
+conjunction with `amazon.aws`.
+
+New modules and plugins developed by the community should be proposed to `community.aws`.
+Content in this collection that is stable and meets other acceptance criteria has the potential
+to be promoted and migrated into `amazon.aws`.
+
+## Submitting Issues
+All software has bugs, and the `amazon.aws` collection is no exception. When you find a bug, 
+you can help tremendously by [telling us about it](https://github.com/ansible-collections/amazon.aws/issues/new/choose).
+
+If you should discover that the bug you’re trying to file already exists in an issue, 
+you can help by verifying the behavior of the reported bug with a comment in that 
+issue, or by reporting any additional information
+
+## Pull Requests
+
+All modules MUST have integration tests for new features. Upgrading to boto3 shall be considered a feature request.  
+Bug fixes for modules that currently have integration tests SHOULD have tests added.  
+New modules should be submitted to the [community.aws](https://github.com/ansible-collections/community.aws) collection
+and MUST have integration tests.
+
+Expected test criteria:
+* Resource creation
+* Resource creation again (idempotency)
+* Resource creation under check mode
+* Resource modification
+* Resource modification again (idempotency)
+* Resource modification under check mode
+* Resource deletion
+* Resource deletion under check mode
+
```

Comment:
```suggestion

Where modules have multiple parameters we recommend running through the 4-step modification cycle for each parameter the module accepts, as well as a modification cycle where as most, if not all, parameters are modified at the same time.
```

I also feel like we should mention using module_defaults:
```
- name: Setup AWS Environment
  module_defaults:
    group/aws:
      aws_access_key: "{{ aws_access_key }}"
      aws_secret_key: "{{ aws_secret_key }}"
      security_token: "{{ security_token | default(omit) }}"
      region: "{{ aws_region }}"
  block:
```
But this is getting into the realms of a full "how to write AWS test cases" guide which is better suited to its own document.

Reply:
I feel like the AWS guide is a better place for that but agree it would be good to have that.  I can add another mention/link to the guide here.